libs/sdk-ui-kit/src/List/Listbox.tsx

```diff
@@ -0,0 +1,238 @@
+            }
+
+            const itemIndex = items.findIndex(
+                (item, index) =>
+                    index > focusedIndex &&
+                    item.stringTitle.toLowerCase().startsWith(char) &&
+                    !item.isDisabled,
+            );
+
+            if (itemIndex !== -1) {
+                setFocusedIndex(itemIndex);
+                return;
+            }
+
+            // If not found after the current index, start from the beginning
+            const fromStartIndex = items.findIndex(
+                (item) => item.stringTitle.toLowerCase().startsWith(char) && !item.isDisabled,
+            );
+
+            if (fromStartIndex !== -1) {
+                setFocusedIndex(fromStartIndex);
+            }
+        },
+        [focusedIndex, items],
+    );
+
+    const handleSelectItem = React.useCallback(
+        (item?: IListboxItem<T>) => {
+            if (!item || item.isDisabled) {
+                return;
+            }
+            onSelect?.(item);
+            onClose?.();
+        },
+        [onClose, onSelect],
+    );
+
+    const handleKeyDown = React.useCallback(
+        (event: React.KeyboardEvent<HTMLUListElement>) => {
+            switch (event.code) {
+                case "ArrowDown":
+                    event.preventDefault();
+                    setFocusedIndex((prevIndex) => {
+                        let newIndex = prevIndex + 1;
+                        // Skip disabled items
+                        while (newIndex < items.length && items[newIndex].isDisabled) {
+                            newIndex++;
+                        }
+                        return newIndex < items.length ? newIndex : prevIndex;
+                    });
+                    break;
+                case "ArrowUp":
+                    event.preventDefault();
+                    setFocusedIndex((prevIndex) => {
+                        let newIndex = prevIndex - 1;
+                        // Skip disabled items
+                        while (newIndex >= 0 && items[newIndex].isDisabled) {
+                            newIndex--;
+                        }
+                        return newIndex >= 0 ? newIndex : prevIndex;
+                    });
+                    break;
+                case "Home":
+                    event.preventDefault();
+                    setFocusedIndex(0);
+                    break;
+                case "End":
+                    event.preventDefault();
+                    setFocusedIndex(items.length - 1);
+                    break;
+                case "Enter":
+                case "Space":
+                    event.preventDefault();
+                    handleSelectItem(items[focusedIndex]);
+                    break;
+                case "Escape":
+                    event.preventDefault();
+                    onClose?.();
+                    break;
+                default:
+                    handleCharacterSearch(event.key.toLowerCase());
```

Comment:
can you add comment that search by typing is here because it is recommended by accessibility best practices?